client/src/store/new-request-module.ts

```diff
@@ -2041,166 +2048,169 @@ export class NewRequestModule extends VuexModule {
       this.mutateDisplayedComponent('ExistingRequestDisplay')
     }
 
-    if (!existingNr) {
+    if (!nrData) {
       handleEmptyResults()
     } else {
-      handleResults(existingNr)
+      handleResults(nrData)
     }
   }
+
   @Action
   async getStats () {
     try {
       let resp = await axios.get('/statistics')
       this.mutateStats(resp.data)
     } catch (error) {
-      // eslint-disable-next-line
-      console.log(error)
+      console.error('getStats() =', error) // eslint-disable-line no-console
     }
   }
+
   @Action
   async checkoutNameRequest (): Promise<boolean> {
     try {
       const { nrId } = this
-      try {
-        const checkedOutBy = sessionStorage.getItem('checkedOutBy')
-        const checkedOutDt = sessionStorage.getItem('checkedOutDt')
-
-        let response
-        if (checkedOutBy) {
-          response = await axios.patch(`/namerequests/${nrId}/checkout`, {
-            checkedOutBy: checkedOutBy,
-            checkedOutDt: checkedOutDt
-          }, {
-            headers: {
-              'Content-Type': 'application/json'
-            }
-          })
-        } else {
-          response = await axios.patch(`/namerequests/${nrId}/checkout`, {}, {
-            headers: {
-              'Content-Type': 'application/json'
-            }
-          })
-        }
-
-        const data = response.data || { checkedOutBy: null, checkedOutDt: null }
-        sessionStorage.setItem('checkedOutBy', data.checkedOutBy)
-        sessionStorage.setItem('checkedOutDt', data.checkedOutDt)
-        return true
-      } catch (err) {
-        await handleApiError(err, 'Could not check out the name request')
```

Comment:
The reason for the nested try-catch block in the old code is this line of code: it always throws an error -- so it needs to be handled.

I refactored the code a bit, but same functionality...